src/core/server/integration_tests/saved_objects/migrations/active_delete.test.ts

```diff
@@ -0,0 +1,329 @@
+          const complexDocuments = indexContents.hits.hits.filter(
+            (result) => result._source?.type === 'complex'
+          );
+
+          expect(complexDocuments.length).toEqual(2);
+          expect(complexDocuments[0]._source).toEqual(
+            expect.objectContaining({
+              complex: {
+                name: 'complex-baz',
+                value: 2,
+              },
+              type: 'complex',
+            })
+          );
+          expect(complexDocuments[1]._source).toEqual(
+            expect.objectContaining({
+              complex: {
+                name: 'complex-lipsum',
+                value: 3,
+              },
+              type: 'complex',
+            })
+          );
+        });
+      });
+    });
+
+    describe('and discardUnknownObjects = false', () => {
+      beforeAll(async () => {
+        esClient = await createBaseline();
+      });
+      afterAll(async () => {
+        await esClient?.indices.delete({ index: `${kibanaIndex}_${currentVersion}_001` });
+      });
+      beforeEach(async () => {
+        await fs.unlink(logFilePath).catch(() => {});
+      });
+
+      it('fails if unknown documents exist', async () => {
+        // remove the 'deprecated' type from the mappings, so that it is considered unknown
+        const types = baselineTypes.filter((type) => type.name !== 'deprecated');
+        const { migrator } = await getKibanaMigratorTestKit({
+          kibanaIndex,
+          types,
+          kibanaVersion: nextMinor,
+          logFilePath,
+        });
+
+        migrator.prepareMigrations();
+
+        try {
+          await migrator.runMigrations();
+        } catch (err) {
+          const errorMessage = err.message;
+          expect(errorMessage).toMatch(
+            'Unable to complete saved object migrations for the [.kibana_migrator_tests] index: Migration failed because some documents were found which use unknown saved object types:'
+          );
+          expect(errorMessage).toMatch(
+            'To proceed with the migration you can configure Kibana to discard unknown saved objects for this migration.'
+          );
+          expect(errorMessage).toMatch(/deprecated:.*\(type: "deprecated"\)/);
+        }
+
+        const logs = await fs.readFile(logFilePath, 'utf-8');
+        expect(logs).toMatch('[.kibana_migrator_tests] INIT -> WAIT_FOR_YELLOW_SOURCE');
+        expect(logs).toMatch(
+          '[.kibana_migrator_tests] WAIT_FOR_YELLOW_SOURCE -> CLEANUP_UNKNOWN_AND_EXCLUDED'
+        );
+        expect(logs).toMatch('[.kibana_migrator_tests] CLEANUP_UNKNOWN_AND_EXCLUDED -> FATAL');
+      });
+
+      it('proceeds if there are no unknown documents', async () => {
+        const { migrator, client } = await getKibanaMigratorTestKit({
+          kibanaIndex,
+          types: baselineTypes,
+          kibanaVersion: nextMinor,
+          logFilePath,
+        });
+
+        migrator.prepareMigrations();
+        await migrator.runMigrations();
```

Comment:
nit: Maybe the toolkit migrator can encapsulate this `prepareMigrations` and `runMigrations` into one `runMigrations` call?